### ISSUE TITLE
Updates to removeEmptyFields

### ISSUE THREAD

**Issue by seanmalbert** [bloom-housing/bloom#2028]
Below was a comment from another PR. We want to cleanup `removeEmptyFields` to not clear out fields when it shouldn't and make sure we are saving things dates and addresses properly.

For rental assistance, if the listing isn't published and I save it (without publishing) when the field is cleared out, it doesn't error, but the change doesn't take affect. The change does work if there is a value in the field.

`removeEmptyFields` is the cause of this and I know in a previous PR the fix was to pass in `keysToIgnore`. That sounded good, but now I'm seeing that we'll have to pass in a lot of keys. If I remember correctly `removeEmptyFields` was introduced due to objects like building address and wanting to know if that was empty. I think it may be the case that we can ignore all non object root level keys like `applicationPickUpAddressType` and `rentalAssistance`, so we don't have to maintain a list of fields to ignore.

It may still be valid to have a `keysToIgnore`, and I think we can add another non-optional key, something like `ignoreRoot`. So from the listing form, when we call `removeEmptyFields`, we pass in true, for `ignoreRoot` and subsequent calls within would pass false. This way we're only checking keys within objects like building address or units. Let me know if you want to discuss this further before implementing.

_Originally posted by @seanmalbert in https://github.com/bloom-housing/bloom/pull/1962#pullrequestreview-777908544_

**Comment by slowbot** [comment]
@emilyjablonski Are there specific review instructions on the front end?

**Comment by emilyjablonski** [comment]
@slowbot Unfortunately no, just the above to test validation flows. This issue changed how we're cleaning the data we send, so all validations.

**Comment by kathyccheng** [comment]
@slowbot I'll take this one!

**Comment by emilyjablonski** [comment]
@kathyccheng Not sure what this one is waiting on?

**Comment by kathyccheng** [comment]
This one is good, closing
